mlir/lib/Target/LLVMIR/ModuleImport.cpp

```diff
@@ -1715,6 +1717,10 @@ void ModuleImport::processFunctionAttributes(llvm::Function *func,
 
   if (func->hasFnAttribute("aarch64_pstate_za_new"))
     funcOp.setArmNewZa(true);
+  if (func->hasFnAttribute("aarch64_pstate_za_preserved"))
+    funcOp.setArmPreservesZa(true);
+  if (func->hasFnAttribute("aarch64_pstate_za_shared"))
+    funcOp.setArmSharedZa(true);
```

Comment:
Maybe slightly rewrite this to show new/shared are mutually exclusive, and `za_preserved` can be used with either.
This would closer match the streaming mode handling above this. 

```suggestion
  if (func->hasFnAttribute("aarch64_pstate_za_new"))
    funcOp.setArmNewZa(true);
  else if (func->hasFnAttribute("aarch64_pstate_za_shared"))
    funcOp.setArmSharedZa(true);    
  // za_preserved can be used with za_new or za_shared. 
  if (func->hasFnAttribute("aarch64_pstate_za_preserved"))
    funcOp.setArmPreservesZa(true);

```